packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { Construct, IResource, Lazy, Resource, Stack, Token, Duration } from '@aws-cdk/core';
+import { CfnDistribution } from './cloudfront.generated';
+import { Origin } from './origin';
+import { CacheBehavior } from './private/cache-behavior';
+
 /**
  * Interface for CloudFront distributions
  */
-export interface IDistribution {
+export interface IDistribution extends IResource {
+  /**
+   * The domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The distribution ID for this distribution.
+   *
+   * @attribute
+   */
+  readonly distributionId: string;
+}
+
+/**
+ * Attributes used to import a Distribution.
+ *
+ * @experimental
+ */
+export interface DistributionAttributes {
   /**
-   * The domain name of the distribution
+   * The generated domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
    */
   readonly domainName: string;
 
   /**
    * The distribution ID for this distribution.
+   *
+   * @attribute
    */
   readonly distributionId: string;
+}
+
+/**
+ * Properties for a Distribution
+ *
+ * @experimental
+ */
+export interface DistributionProps {
+  /**
+   * The default behavior for the distribution.
+   */
+  readonly defaultBehavior: BehaviorOptions;
+
+  /**
+   * Additional behaviors for the distribution, mapped by the pathPattern that specifies which requests to apply the behavior to.
+   *
+   * @default no additional behaviors are added.
+   */
+  readonly additionalBehaviors?: Record<string, BehaviorOptions>;
+
+  /**
+   * A certificate to associate with the distribution. The certificate must be located in N. Virginia (us-east-1).
+   *
+   * @default the CloudFront wildcard certificate (*.cloudfront.net) will be used.
+   */
+  readonly certificate?: acm.ICertificate;
+
+  /**
+   * The price class that corresponds with the maximum price that you want to pay for CloudFront service.
+   * If you specify PriceClass_All, CloudFront responds to requests for your objects from all CloudFront edge locations.
+   * If you specify a price class other than PriceClass_All, CloudFront serves your objects from the CloudFront edge location
+   * that has the lowest latency among the edge locations in your price class.
+   *
+   * @default PRICE_CLASS_ALL
+   */
+  readonly priceClass?: PriceClass;
+
+  /**
+   * How CloudFront should handle requests that are not successful (e.g., PageNotFound).
+   *
+   * @default - No custom error configuration.
+   */
+  readonly errorConfigurations?: CustomErrorResponse[];
```

Comment:
As a rule, don't use the term `configuration` in L2 props. Everything is configuration.

I would call this `errorResponses` or something like that.

Reply:
Do we need the type name to be called `CustomErrorResponse`? What's custom about it? Would `ErrorResponse` be sufficient?

Reply:
`ErrorResponse` is fine, and I can rename the property `errorResponses`.